CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/clipboard/CopyToClipboardAction.java

```diff
@@ -54,13 +56,24 @@ public void actionPerformed(final ActionEvent e) {
         final Graph graph = context.getGraph();
 
         // TODO: make this a SimpleReadPlugin.
-        PluginExecution.withPlugin(new SimplePlugin("Copy To Clipboard") {
-            @Override
-            protected void execute(final PluginGraphs graphs, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
-                final Graph graph = graphs.getGraph();
-                PluginExecution.withPlugin(InteractiveGraphPluginRegistry.COPY).executeNow(graph);
+        PluginExecution.withPlugin(new CopyToClipboard()).executeLater(graph);
+    }
+
+    /**
+     * Plugin to copy to the clipboard.
+     */
+    @PluginInfo(pluginType = PluginType.EXPORT, tags = {"EXPORT"})
+    private class CopyToClipboard extends SimplePlugin {
+
+        @Override
+        public String getName() {
+            return "Copy To Clipboard Action: Copy To Clipboard";
+        }
 
-            }
-        }).executeLater(graph);
+        @Override
+        protected void execute(final PluginGraphs graphs, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
+            final Graph graph = graphs.getGraph();
+            PluginExecution.withPlugin(InteractiveGraphPluginRegistry.COPY).executeNow(graph);
```

Comment:
This is just more of an aside but it almost feels like this plugin is unnecessary. All it's doing is executing another plugin on a graph and it appears there is a graph gotten earlier on in the action (the one from the context). Is there something wrong with that graph that means we need to grab the graph differently here? @arcturus2

Reply:
No there isn't @antares1470 and we could actually replace this to extend `SimplePluginAction`. I will have a look at doing this after the RC2 is done.